sdk/ai/Azure.AI.Inference/samples/Sample8_ChatCompletionsWithTelemetry.md

```diff
@@ -0,0 +1,145 @@
+# Using telemetry with Azure.AI.Inference
+
+In this example we will demonstrate how [OpenTelemetry](https://learn.microsoft.com/dotnet/core/diagnostics/observability-with-otel) library can be leveraged to diagnose the issues in the Chat completion request.
+
+## Project creation and dependency installation.
+First, we will create the console application project and add `Azure.AI.Inference` as a dependency. The first command will create the project called `TelemetryDemo.csproj`. The ` dotnet add package <…>` command will modify project file and in future we will need to run `dotnet restore` to install all dependencies, if we will remove the installed ones.
+
+```bash
+dotnet new console --name TelemetryDemo --output TelemetryDemo
+dotnet add package Azure.AI.Inference --prerelease
+```
+
+Now we will need to add the dependencies for OpenTelemetry.
+
+```bash
+dotnet add package OpenTelemetry
+dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package Azure.Monitor.OpenTelemetry.AspNetCore --prerelease
+```
+
+## Create the simple application with telemetry.
+The `dotnet new` created the project with the single file called Program.cs. Let us edit this file with the IDE of choice.
+
+First we will import open telemetry and Azure.AI.Inference
+```c#
+//Azure imports
+using Azure;
+using Azure.AI.Inference;
+// Open telemetry imports
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using OpenTelemetry.Metrics;
+```
+
+Now we will define the `endpoint`, `credential` and `model`. In this example we will take these values from the environment variables.
+```c#
```

Comment:
All of the C# code in the samples must be snippets.  See: [Updating Sample Snippets](https://github.com/Azure/azure-sdk-for-net/tree/main/CONTRIBUTING.md#updating-sample-snippets)

Reply:
Fixed. The code was re-generated from the samples test.